pkg/steps/project_image.go

```diff
@@ -64,6 +72,42 @@ func (s *projectDirectoryImageBuildStep) run(ctx context.Context) error {
 	return handleBuild(ctx, s.client, build)
 }
 
+func (s *projectDirectoryImageBuildStep) createSecrets() error {
```

Comment:
You caught me. I copy-and-modify. :)
Will give it a try.

Reply:
Tried to abstract the logic but did not succeed.
The hard part is collecting `toCreates` because the source secrets are from different fields of ci-op's config.
Once we have `toCreates` there is nothing left except a simple Create call on the client.

Reply:
Why don't we write

```go
func CopySecretsIntoJobNamespace(ctx context.Context, jobSpec JobSpec, secrets []ctrlruntimeclient.ObjectKey) error {}
```

?

Reply:
Updated.